### PR TITLE
refactor: bring autoswap up to date on the new Zoe spike branch

### PR DESCRIPTION
**Pull request by Chris-Hibbert**
This brings the autoswap contract up-to-date on the new-zoe-spike-2 branch, based on the synchronous seat work.

The unit tests for autoswap pass. I haven't yet done the swingset tests.

### CODE REVIEW

File: packages/zoe/src/contracts/autoswap.js [hunk 1]
```diff
@@ -23,245 +22,233 @@ import '../../exported';
...
+      if (brand === brandToMatch) {
+        return keyword;
       }
-      throw new Error('getPoolKeyword: brand not found');
-    };
+    }
+    throw new Error('getPoolKeyword: brand not found');
+  };
+
+  const { zcfSeat: poolSeat } = await zcf.addEmptySeat();
```
Comment (katelynsills) [inline packages/zoe/src/contracts/autoswap.js]: is this meant to be async? If so, why not use makeEmptyOffer?

File: packages/zoe/src/contracts/autoswap.js [hunk 2]
```diff
@@ -23,245 +22,233 @@ import '../../exported';
...
+  const publicFacet = harden({
+    getCurrentPrice,
+    getLiquidityIssuer: () => liquidityIssuer,
+    getPoolAllocation,
+    makeSwapInvite,
+    makeAddLiquidityInvite,
+    makeRemoveLiquidityInvite,
   });
+
+  return { publicFacet, creatorFacet: publicFacet };
```
Comment (katelynsills) [inline packages/zoe/src/contracts/autoswap.js:248]: Ooo nice, I like the concept of passing the publicFacet back as the creatorFacet. We might consider always passing the publicFacet back to the creator too.

**Review by katelynsills — APPROVED** [review]
Looks good! I need to learn more about how `addEmptySeat` works - it was surprising to me that it was asynchronous.

**Comment by Chris-Hibbert** [comment]
addresses #1267, for autoswap. I think that bug also refers to multipoolAutoswap, so I won't close it.

File: packages/zoe/src/contracts/autoswap.js (hunk 1 above)
Comment (Chris-Hibbert) [reply packages/zoe/src/contracts/autoswap.js]: Dropped the `async`. Thanks.

**Comment by Chris-Hibbert** [comment]
>  I need to learn more about how addEmptySeat works - it was surprising to me that it was asynchronous.

No, it was my oversight.

**Pull request merged**